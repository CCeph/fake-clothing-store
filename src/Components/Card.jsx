import PropTypes from "prop-types";

export default function Card({ image, price, name }) {
  return (
    <div data-testid="shop-item">
      <img src={image} alt="Insert image here" />
      <p>${price}</p>
      <p>{name}</p>
    </div>
  );
}

Card.propTypes = {
  image: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};
