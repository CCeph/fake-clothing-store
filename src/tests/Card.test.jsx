import { render } from "@testing-library/react";
import Card from "../Components/Card";
import { expect } from "vitest";
import { MemoryRouter } from "react-router-dom";

it("renders card properly", () => {
  const price = 0;
  const name = "name";
  const { container } = render(
    <MemoryRouter>
      <Card
        image={"https://i.imgur.com/cBuLvBi.jpeg"}
        price={price}
        name={name}
      />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
