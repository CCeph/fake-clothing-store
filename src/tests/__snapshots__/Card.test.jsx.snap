// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders card properly 1`] = `
<div>
  <div
    data-testid="shop-item"
  >
    <img
      alt="Insert image here"
      src="/src/assets/urban.png"
    />
    <p>
      $
      0
    </p>
    <p>
      name
    </p>
  </div>
</div>
`;
